h2/src/main/org/h2/expression/analysis/WindowFrame.java

```diff
@@ -302,17 +303,47 @@ private static int getIntOffset(WindowFrameBound bound, Value[] values, Session
         OpType opType = add ^ (sortOrder.getSortTypes()[0] & SortOrder.DESCENDING) != 0 ? OpType.PLUS : OpType.MINUS;
         Value[] row = orderedRows.get(currentRow);
         Value[] newRow = row.clone();
-        newRow[sortIndex] = new BinaryOperation(opType, //
-                ValueExpression.get(row[sortIndex]),
-                ValueExpression.get(getValueOffset(bound, orderedRows.get(currentRow), session))) //
-                        .optimize(session).getValue(session);
+        Value currentValue = row[sortIndex];
+        switch (currentValue.getType()) {
+        case Value.BYTE:
+        case Value.SHORT:
+        case Value.INT:
+        case Value.LONG:
+        case Value.DECIMAL:
+        case Value.DOUBLE:
+        case Value.FLOAT:
+        case Value.TIME:
+        case Value.DATE:
+        case Value.TIMESTAMP:
+        case Value.TIMESTAMP_TZ:
+        case Value.INTERVAL_YEAR:
+        case Value.INTERVAL_MONTH:
+        case Value.INTERVAL_DAY:
+        case Value.INTERVAL_HOUR:
+        case Value.INTERVAL_MINUTE:
+        case Value.INTERVAL_SECOND:
+        case Value.INTERVAL_YEAR_TO_MONTH:
+        case Value.INTERVAL_DAY_TO_HOUR:
+        case Value.INTERVAL_DAY_TO_MINUTE:
+        case Value.INTERVAL_DAY_TO_SECOND:
+        case Value.INTERVAL_HOUR_TO_MINUTE:
+        case Value.INTERVAL_HOUR_TO_SECOND:
+        case Value.INTERVAL_MINUTE_TO_SECOND:
+            break;
+        default:
+            throw DbException.getInvalidValueException("ORDER BY value for RANGE frame", currentValue.getTraceSQL());
```

Comment:
"unsupported type for ORDER value for RANGE frame:

Reply:
I used a sort key term here for consistency with another message.

Reply:
ok